### PR TITLE
Variable protected names

### LINKED ISSUE #17457: Prevent creating variables that are flux keywords
Due to a  bug in the flux parser this will cause issues.

### PR DESCRIPTION
**Pull request by zoesteinkamp**
Closes #17457 
Screen Shot 2020-07-10 at 10 45 42 AM

Describe your proposed changes here.

- [ ] [CHANGELOG.md](https://github.com/influxdata/influxdb/blob/master/CHANGELOG.md) updated with a link to the PR (not the Issue)
- [ ] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [ ] Rebased/mergeable
- [ ] Tests pass
- [ ] http/swagger.yml updated (if modified Go structs or API)
- [ ] Documentation updated or issue created (provide link to issue/pr)
- [ ] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

File: variable.go [hunk 1]
```diff
@@ -127,6 +127,16 @@ func (m *Variable) Valid() error {
 		return fmt.Errorf("invalid arguments type")
 	}
 
+	inValidNames := [11]string{"and", "import", "not", "return", "option", "test", "empty", "in", "or", "package", "builtin"}
+
+	for x := 0; x < len(inValidNames); x++ {
```
Comment (glinton) [inline variable.go:132]: While this is valid, it's not preferred.
```suggestion
	for x := range inValidNames {
```

File: variable.go [hunk 2]
```diff
@@ -127,6 +127,16 @@ func (m *Variable) Valid() error {
...
+		fmt.Printf(inValidNames[x])
```
Comment (glinton) [inline variable.go:133]: remove this print prior to merge

File: variable.go [hunk 3]
```diff
@@ -127,6 +127,16 @@ func (m *Variable) Valid() error {
...
+
+		if m.Name == inValidNames[x] {
+			return fmt.Errorf("protected variable name")
```
Comment (glinton) [inline variable.go:136]: return the name of the variable for clarity perhaps?
```suggestion
			return fmt.Errorf("%q is a protected variable name", inValidNames[x])
```

Comment (zoesteinkamp) [reply variable.go:136]: Added it!

File: variable.go (hunk 2 above)
Comment (zoesteinkamp) [reply variable.go:133]: Good catch, removed

File: variable.go (hunk 1 above)
Comment (zoesteinkamp) [reply variable.go:132]: got it, i changed it

**Pull request merged**